src/transformers/keras_callbacks.py

```diff
@@ -169,9 +192,13 @@ def on_epoch_end(self, epoch, logs=None):
                 predictions = self.model.generate(generation_inputs, attention_mask=attention_mask)
             else:
                 predictions = self.model.predict(batch)
-            predictions = dict(predictions)
+            if isinstance(predictions, dict):
+                # BatchEncoding is a subclass of dict
+                predictions = dict(predictions)
```

Comment:
I need a comment here that will stop me from thinking you're gone crazy ;-)

(I think I know why you're doing that, but not quite sure.)

Reply:
The answer is that Keras does some *very* weird things if you pass it a `BatchEncoding`, or other dict-subclass. It has to be a `dict`, for some reason. But yes, I should explain it in the comments!